src/preamble.js

```diff
@@ -430,45 +430,55 @@ function stringToAscii(str, outPtr) {
   return writeAsciiToMemory(str, outPtr, false);
 }
 
-// Given a pointer 'ptr' to a null-terminated UTF8-encoded string in the given array that contains uint8 values, returns
-// a copy of that string as a Javascript String object.
-
+// Given a pointer 'ptr' to an UTF8-encoded string (optionally null terminated) in the given array that contains uint8 values,
+// returns a copy of that string as a Javascript String object.
+// sizeInBytes parameter (optional): specify size of bytes to be converted in the array. When sizeInBytes is used,
+//                                   null bytes inside or at the end of the buffer are not interpreted as terminators.
+//                                   Note that partial/incomplete UTF-8 sequences at the end are discarded.
```

Comment:
Before the EMBIND_STD_STRING_IS_UTF8 switch was available, the sizeInBytes parameter has been added to the UTF8ArrayToString() method in order to support inline null bytes in the input data that should not be considered as terminators. Nowadays, we explicitly differentiate between UTF-8 and (binary safe) non UTF-8 string mapping mode. Provided that we consider UTF-8 mode as not binary safe, the optional sizeInBytes can be removed again. There is no relation to error handling.

Published a change for the removal of sizeInBytes (see 2d4c7e5).

Reply:
Well, as unit testing shows ;-), the sizeInBytes parameter at UTF8ArrayToString() is not as obsolete as I had in mind. While it is not needed for error handling, it is still required for embind itself.
Within embind.js during string conversion (std_sting>registerType>fromWireType), the string data on the heap is not always(/never?) in null-terminated form. 
Therefore we need a suitable conversion function with an externally defined length instead of solely relying on null-termination.

Reverted 2d4c7e5 because of this issue.